azure-pipeline/src/index.ts

```diff
@@ -0,0 +1,113 @@
+    // a function takes user's input given in the .yml file
+    // and request an bearer token for authentication of later authorization
+    // using @azure/identity
+    private getAuthorizationTokenFromKey = async () => {
+        try {
+            const params = {
+                tenantID:  validatedAuthParams('tenantid'),
+                clientID:  validatedAuthParams('servicePrincipalId'),
+                servicePrincipalKey: validatedAuthParams('servicePrincipalKey'),
+            }
+            const clientSecretCredential: ClientSecretCredential = new ClientSecretCredential(
+                params.tenantID, params.clientID, params.servicePrincipalKey
+            );
+            const accessToken = await clientSecretCredential.getToken(SCOPE);
+            if (!accessToken) {
+                throw new AuthenticationError(0, {
+                    error: 'Access token fetch failed'
+                });
+            }
+            return accessToken;
+        } catch (e) {
+            throw new Error("Authorization parameters validation failed");   
+        }
+    }
+
+    // getConvertedFileObject():
+    // compress config file folder
+    // and make it ready for sending it to backend
+    private getConvertedFileObject = async () => {
+        const file = await this.f.compressFile(
+            validatedEnvVar(INPUT.source),
+            validatedEnvVar(INPUT.target),
+            OUTPUT_PATH,
+        );
+        return {
+            properties: {
+                rootFile: `${validatedEnvVar(INPUT.target)}nginx.conf`,
+                package: {
+                    data: this.f.convertFileToBase64String(file),
+                }
+            }
+        };
+    }
+
+    // getRequestConfig():
+    // make up the bearer token from user's input
+    private getRequestConfig = async () => {
+        const token = await this.getAuthorizationTokenFromKey();
+        const config: AxiosRequestConfig = { headers: {
+            Authorization: `Bearer ${token.token}`,
+        }}
+        return config;
+    }
+
+    // getRequestResource():
+    // make up all needed parameters for the uploading api request
+    private getRequestResource = () => {
+        return {
+            subscriptionId: validatedEnvVar(INPUT.subscription),
+            resourceGroupName: validatedEnvVar(INPUT.resource),
+            deploymentName: validatedEnvVar(INPUT.deployment),
+            apiVersion: API_VERSION,
+        }
+    }
+
+    // updateNginxConfig():
+    // main function to compress file, get authentication info, and call the api
+    updateNginxConfig = async () => {
+        console.log('updateNginxConfig...')
+        try {
+            const res = await updateUserConfig(
+                this.getRequestResource(),
+                await this.getConvertedFileObject(),
+                await this.getRequestConfig(),
+            );
+            console.log('Nginx config successfully uploaded!\n');
```

Comment:
```suggestion
            console.log('Nginx config successfully uploaded!');
```

Is it usual to use `console.log` to print to stdout or whatever? In the past I've used packages that would allow the caller to enabled/disable output where it doesn't suit them, but I think this is not meant to be consumed as a library.